src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -733,18 +1120,15 @@ InputValues* CompositeClosure::do_add_closure(
     // Make sure we have enough space.
     if (get_num_closures() >= MaxClosureEntries)
     {
-        RENDERER_LOG_WARNING("maximum number of closures in osl shader group exceeded; ignoring closure.");
-        return 0;
+        throw ExceptionOSLRuntimeError(
+            "Maximum number of closures in OSL shader group exceeded");
     }
 
     assert(m_num_bytes + sizeof(InputValues) <= MaxPoolSize);
 
     // We use the luminance of the weight as the BSDF weight.
     const double w = luminance(weight);
-
-    // Ignore zero or negative weights.
-    if (w <= 0.0)
-        return 0;
+    assert (w > 0.0);
```

Comment:
Is w going to always be strictly positive, regardless of user input? If not, then an assertion is not the right solution.